src/Control/Email/Email.php

```diff
@@ -276,14 +276,26 @@ public function setSwiftMessage($swiftMessage)
         $dateTime = new DateTime();
         $dateTime->setTimestamp(DBDatetime::now()->getTimestamp());
         $swiftMessage->setDate($dateTime);
-        if (!$swiftMessage->getFrom() && ($defaultFrom = $this->config()->get('admin_email'))) {
-            $swiftMessage->setFrom($defaultFrom);
+        if (!$swiftMessage->getFrom()) {
+            $swiftMessage->setFrom($this->getDefaultFrom());
         }
         $this->swiftMessage = $swiftMessage;
 
         return $this;
     }
 
+    /**
+     * @return string
+     */
+    private function getDefaultFrom(): string
+    {
+        $defaultFrom = $this->config()->get('admin_email');
+        if ($defaultFrom) {
+            return $defaultFrom;
+        }
+        return sprintf('no-reply@%s', parse_url(Director::host(), PHP_URL_HOST));
```

Comment:
Can we please add an extension point `updateDefaultFrom` before the method returns? It will be useful for CWP sites migrating to AWS with subsites where we need to be able to update the default email address based on the selected subsite. The extension point here would allow for that.

Reply:
Have added extension point

Reply:
This is targeted to a patch release. We shouldn't be adding hooks. 

And as previously mentioned, this is meant to be a stop gap measure for people who forgot to explicitly define a default admin email ... not a new feature.